base_layer/p2p/src/services/executor.rs

```diff
@@ -64,30 +64,35 @@ impl ServiceExecutor {
 
         let mut senders = Vec::new();
 
-        for mut service in registry.services.into_iter() {
+        for service in registry.services.drain(..) {
```

Comment:
I believe into_iter would be better here. 
Into_iter moves the vec, while drain mutably borrows it. 

Because you are moving the whole struct in and not reusing it, moving it should be better

Reply:
Pretty sure its functionally the same thing once the struct is moved into the thread. This was part of trying to fix a bug but didn't see the difference leaving it like this.